pymeasure/instruments/philips/PM6669.py

```diff
@@ -0,0 +1,202 @@
+
+        If a request is made to the device while a measurement is ready, both the reply and the
+        measurement are returned. The measurement is filtered out and put in a backlog Queue.
+        """
+        reply = ''
+        result = None
+        while reply == '':
+            reply = super().read()
+            reply = reply.strip('\x00')
+            for line in reply.splitlines():
+                if line[:3] in self.KEYWORDS:
+                    self.backlog.put([datetime.now(), line])
+                elif line != '':
+                    result = line
+        return result
+
+
+PM6669.id = Instrument.measurement(
+    "ID?", """ Reads the instrument identification """
+)
+
+PM6669.function = Instrument.control(
+    "FNC?", "%s", """A string or keyowrd property that sets measuring function on the device.""",
+    validator=strict_discrete_set,
+    values={"FREQ A": "FREQ A", "FREQ B": "FREQ B", "RPM A": "RPM A", "PER A": "PER A",
+            "WIDTH A": "WIDTH A", "TOTM A": "TOTM A",
+            Functions.FREQUENCY_A: "FREQ   A",
+            Functions.FREQUENCY_B: "FREQ   B", Functions.PER_A: "PER    A",
+            Functions.RPM_A: "RPM    A",
+            Functions.WIDTH_A: "PWIDTH A", Functions.TOT_A: "TOTM   A"
+            },
+    map_values=True
+)
+
+PM6669.gate_open = Instrument.measurement(
+    "GATE OPEN", """Open the gate and return the current count""",
+    get_process=lambda x: float(x[6:])
+)
+
+PM6669.gate_close = Instrument.measurement(
+    "GATE CLOSE", """Close the gate and return the current count""",
+    get_process=lambda x: float(x[6:])
+)
+
+PM6669.measurement_time = Instrument.control(
+    "MEAC?", "MTIME %g", """ A float property that controls the measurement time""",
+    validator=strict_range,
+    values=[0, 10],
+    get_process=lambda x: float(x[0][5:]) if x[0].startswith("MTIME") is True else 0
+)
+
+PM6669.freerun = Instrument.control(
+    "MEAC?", "FRUN %s", """ A boolean property that controls the freerun settings""",
+    validator=strict_discrete_set,
+    values={True: "ON", False: "OFF"},
+    map_values=True,
+    get_process=lambda x:
+    (x[1].split("\n")[0][5:] == " ON") if x[0].startswith("MTIME") is True else 0
+)
+
+PM6669.timeout = Instrument.control(
+    "MEAC?", "TOUT %s",
+    """ A float property that controls the measurement timeout, this timeout only has meaning when
+        freerun is off.""",
+    validator=strict_range,
+    values=[0, 25.5],
+    get_process=lambda x: float(x[1].split("\n")[2][5:]) if x[0].startswith("MTIME") is True else 0
+)
+
+PM6669.SRQMask = Instrument.control(
+    "BUS?", "MSR %i",
+    """A integer property that controls the SRQ mask""",
+    get_process=lambda x: MSRFlag(int(x[0].split(",")[0].split(" ")[-1]))
+)
+
+PM6669.meac = Instrument.measurement(
+    "MEAC?", """ Reads the measurement settings from the device """
+)
+
+PM6669.defaults = Instrument.measurement(
+    "DCL", """ Reset to instrument defaults """
```

Comment:
"Reset to.." implies an action, but this only reads what comes back from the instrument (I guess nothing in this case). I think this should be a method `reset_to_defaults(self)` that used `self.write("DCL")`.